pkg/documents/documents.go

```diff
@@ -240,12 +239,15 @@ func retrieveRawDocument(logger log.Logger, repo DocumentRepository, keeper *sec
 		}
 		defer rdr.Close()
 
-		encryptedDoc, err := ioutil.ReadAll(rdr)
-		if err != nil {
-			logger.Log("documents", "failed reading document from storage bucket", "error", err, "customerID", customerID, "documentID", documentID, "requestID", requestID)
+		encryptedDoc := make([]byte, rdr.Size())
+		if n, err := rdr.Read(encryptedDoc); err != nil || n == 0 {
```

Comment:
Should we check if `n != rdr.Size()`? I think we do a similar `n == 0` check elsewhere with this same `.Read(..)` pattern.

Reply:
Good call. I'm just running the tests, then I'll push that up.